src/main/java/com/tealium/PersistentUdo.java

```diff
@@ -0,0 +1,80 @@
+package com.tealium;
+
+import java.io.IOException;
+
+/**
+ * Tealium data object for processing generation of standardized universal data
+ * points.
+ *
+ * @author Chad Hartman, Jason Koo, Merritt Tidwell, Karen Tamayo, Chris Anderberg
+ */
+class PersistentUdo {
+
+    private TextStorage textStorage;
+
+    public PersistentUdo(TextStorage textStorage) {
+        this.textStorage = textStorage;
+    }
+
+    /**
+     * Read serialized Map from storage.
+     * 
+     * Maps are persisted using percent encoding so as to be dependency
+     * Independent; and immune to the vulnerabilities of the native
+     * serialization API.
+     * 
+     * @return The deserialized map.
+     * @throws PersistentDataAccessException
+     *             If the file doesn't exist or the map is malformed.
+     */
+    public Udo readOrCreateUdo(Udo defaultData) throws UdoSerializationException {
+        Udo loadedUdo = null;
+
+        try {
+            String loadedText = this.textStorage.readText();
```

Comment:
Stack the catch blocks one after another and only use one try.

Reply:
That won't work. The reason they are nested is because the third try block should only execute if the second try block fails, which is why the third try block is nested in the second catch block. Likewise both the second and third try blocks must be nested in the first try block since they both require access to a bit of information (the loaded persistent text) that can itself result in an exception while trying to access.